src/table/src/table_reference.rs

```diff
@@ -0,0 +1,76 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt::{self, Display};
+
+use datafusion_common::TableReference as DfTableReference;
+
+/// Represents a resolved path to a table of the form “catalog.schema.table”
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub struct TableReference<'a> {
+    pub catalog: &'a str,
+    pub schema: &'a str,
+    pub table: &'a str,
+}
+
+pub type OwnedTableReference = TableReference<'static>;
+
+// TODO(LFC): Find a better place for `TableReference`,
+// so that we can reuse the default catalog and schema consts.
+// Could be done together with issue #559.
+impl<'a> TableReference<'a> {
+    pub fn bare(table: &'a str) -> Self {
+        TableReference {
+            catalog: "greptime",
+            schema: "public",
+            table,
+        }
+    }
```

Comment:
I think this "todo" can be removed, and consts can be reused:

```suggestion
impl<'a> TableReference<'a> {
    pub fn bare(table: &'a str) -> Self {
        TableReference {
            catalog: common_catalog::consts::DEFAULT_CATALOG_NAME,
            schema: common_catalog::consts::DEFAULT_SCHEMA_NAME,
            table,
        }
    }
```